### DISCUSSION TITLE
chnage the value of a variable with x-if

### DISCUSSION THREAD

**Discussion by sxshateri** [alpinejs/alpine#858] [4. General]
I'm trying to change the value of the `limitError` once `content.length > limit` is true but seems i can't get how to do it. 
```
<div class="col-md-12 mt-2">
   <div class="form-group" x-data="{ content: '', limit: $el.dataset.limit, limitError: false }" data-limit="20">
       <label class="justify-content-start pr-0" for="aboutMe">درباره من</label>
       <textarea class="form-control form-control-alternative" x-ref="content" x-model="content" rows="4" name="about-me" id="aboutMe" placeholder="در ۲۵۰ کاراکتر توضیحی درباره خود دهید ..."></textarea>
       <small x-ref="remaining" class="text-muted" x-bind:class="{'text-red': limitError}"><span x-text="limit"></span> / <span x-text="limit - content.length" x-if="{content.length > limit : limitError}"></span></small>
   </div>
</div>
```

**Comment by SimoTod** [comment]
https://github.com/alpinejs/alpine#x-if
X-if is only available with template tags.

**Reply by sxshateri** [reply]
I figured out how to do it only with x-bind. it was easier than I thought. I was looking at it from a different angle. thanks for the reply.

**Reply by SimoTod** [reply]
You can also use x-show which is usually simpler than x-if. Same syntax but it applies to any element like your initial example. The difference is that x-show does not remove the element from the page but it just applies display none
